File: module-gan/src/main/java/com/huyingbao/module/gan/ui/random/action/RandomActionCreator.java
package com.huyingbao.module.gan.ui.random.action;

import com.huyingbao.core.arch.action.RxActionCreator;
import com.huyingbao.core.arch.action.RxActionManager;
import com.huyingbao.core.arch.dispatcher.RxDispatcher;
import com.huyingbao.core.arch.model.RxAction;
import com.huyingbao.core.arch.scope.ActivityScope;
import com.huyingbao.module.gan.action.GanApi;
import com.huyingbao.module.gan.action.GanConstants;

import javax.inject.Inject;


/**
 * rxAction创建发送管理类
 * <p>
 * Created by liujunfeng on 2019/1/1.
 */
@ActivityScope
public class RandomActionCreator extends RxActionCreator implements RandomAction {
    private GanApi mGanApi;

    @Inject
    RandomActionCreator(RxDispatcher rxDispatcher, RxActionManager rxActionManager, GanApi ganApi) {
        super(rxDispatcher, rxActionManager);
        mGanApi = ganApi;
    }

    @Override
    public void getDataList(String category, int count, int page) {
        RxAction rxAction = newRxAction(GET_DATA_LIST,
                GanConstants.Key.COUNT, count,
                GanConstants.Key.PAGE, page);
        postHttpAction(rxAction, mGanApi.getDataList(category, count, page));
    }
}

File: module_gan_kotlin/src/main/java/com/huyingbao/module/gan/ui/random/action/RandomActionCreator.kt
package com.huyingbao.module.gan.ui.random.action

import com.huyingbao.core.arch.action.RxActionCreator
import com.huyingbao.core.arch.action.RxActionManager
import com.huyingbao.core.arch.dispatcher.RxDispatcher
import com.huyingbao.core.arch.model.RxAction
import com.huyingbao.core.arch.scope.ActivityScope
import com.huyingbao.module.gan.action.GanApi
import com.huyingbao.module.gan.action.GanConstants

import javax.inject.Inject


/**
 * rxAction创建发送管理类
 * Created by liujunfeng on 2019/1/1.
 */
@ActivityScope
class RandomActionCreator @Inject
internal constructor(rxDispatcher: RxDispatcher, rxActionManager: RxActionManager) : RxActionCreator(rxDispatcher, rxActionManager), RandomAction {
    @Inject
    internal var mGanApi: GanApi? = null

    override fun getProductList(category: String, count: Int, page: Int) {
        val action = newRxAction(RandomAction.GET_PRODUCT_LIST,
                GanConstants.Key.COUNT, count,
                GanConstants.Key.PAGE, page)
        postHttpAction<GanResponse<Product>>(action, mGanApi!!.getDataList(category, count, page))
    }
}
